### ISSUE TITLE
essentials.chat.spy.exempt Does not hide messages sent to that user

### ISSUE THREAD

**Issue by TomLewis** [EssentialsX/Essentials#166]
The purpose of having essentials.chat.spy.exempt for my OP's is to allow mods to monitor all chat apart from what my OP's send/receive, now this half works, it stops social spy showing what these people send back, but it does not stop the reply messages being sent to them in the first place, so they can easily figure out the conversation by having one half oh it open to read!

I want players to be able to talk to my admins and trust their information stay private, especially if we have to ask for personal information like emails etc when there is a payment issue.

**Comment by Kakifrucht** [comment]
This would introduce quiet alot of code, as the chatspy would have to figure out who the actual recipient is. Afaik the chatspy and the message module in Essentials are pretty much sperated, so they would have to make calls to each other. Currently the implementation is just very lightweight and not supposed to block what players write to OPs.

**Comment by drtshock** [comment]
Oh, that's a good point. This would be quite a bit.

**Comment by TomLewis** [comment]
That's a shame I imagined it was done at chat level, so when the message was being sent it could run a quick check.

**Comment by drtshock** [comment]
It's done at the command level since we do it for multiple commands and allow you to add / remove any to the list.

**Comment by SupaHam** [comment]
Maybe we can introduce a socialspy ranking list in the config:

``` yaml
socialspy-ranks:
  admin: ['*']
  mod+: ['default', 'mod']
  mod: ['default']
```

@drtshock @vemacs thoughts?

**Comment by Kakifrucht** [comment]
That doesn't sound very intuitive to me.

**Comment by SupaHam** [comment]
Why not? It's pretty self explanatory. I could go a step further and make ranking be inherited so you can remove the `'default'` from `mod+` because `mod` already has it.

What would you consider more intuitive?

**Comment by Kakifrucht** [comment]
Sorry, don't really have a better idea either.

**Comment by Kakifrucht** [comment]
Why not just add `&& !recipient.isAuthorized("essentials.chat.spy.exempt"))` to [here](https://github.com/drtshock/Essentials/blob/2.x/Essentials/src/com/earth2me/essentials/messaging/SimpleMessageRecipient.java#L83)?

**Comment by SupaHam** [comment]
I'll do that for now, but I wanted to fix #516 at the same time.

**Comment by SupaHam** [comment]
This has been fixed in 017082c. You can grab the fix from [build 413](https://ci.drtshock.net/job/EssentialsX/) on the CI server.
